src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -36,6 +37,11 @@ internal abstract class SynthesizedParameterSymbolBase : ParameterSymbol
             _ordinal = ordinal;
             _refKind = refKind;
             _name = name;
+
+            if (this.RefKind == RefKind.RefReadOnly)
+            {
+                this.DeclaringCompilation.EnsureIsReadOnlyAttributeExists(this);
```

Comment:
I think the caller should be responsible for performing this check, after all we got RefKind.RefReadOnly from somewhere.

---
In reply to: [113566464](https://github.com/dotnet/roslyn/pull/18715#discussion_r113566464) [](ancestors = 113566464)